client/src/components/History/Content/ContentItem.vue

```diff
@@ -0,0 +1,204 @@
+import { Nametag } from "components/Nametags";
+import ContentDescription from "./ContentDescription";
+import DatasetDetails from "./Dataset/DatasetDetails";
+import CONTENTSTATE from "./contentState";
+
+export default {
+    components: {
+        ContentDescription,
+        DatasetDetails,
+        Nametag,
+    },
+    props: {
+        item: { type: Object, required: true },
+        id: { type: Number, required: true },
+        name: { type: String, required: true },
+        state: { type: String, default: null },
+        expanded: { type: Boolean, required: true },
+        selected: { type: Boolean, default: false },
+        expandable: { type: Boolean, default: true },
+        selectable: { type: Boolean, default: false },
+        writeable: { type: Boolean, default: true },
+    },
+    computed: {
+        contentId() {
+            return `dataset-${this.item.id}`;
+        },
+        clsStatus() {
+            const status = CONTENTSTATE[this.state] && CONTENTSTATE[this.state].status;
+            if (this.selected) {
+                return "alert-info";
+            } else if (!status) {
+                return `alert-success`;
+            } else {
+                return `alert-${status}`;
+            }
+        },
+        displayButtonTitle() {
+            if (this.item.purged) {
+                return "Cannot display datasets removed from disk.";
+            }
+            if (this.isUnavailable) {
+                return "This dataset is not yet viewable.";
+            }
+            return "Display";
+        },
+        deleteButtonTitle() {
+            return this.item.purged
+                ? "This dataset has been permanently deleted."
+                : this.item.deleted
+                ? "Undelete"
+                : "Delete";
+        },
+        editButtonTitle() {
+            if (this.item.deleted) {
+                return "Undelete this dataset to edit attributes.";
+            }
+            if (this.item.purged) {
+                return "Cannot edit attributes of datasets removed from disk.";
+            }
+            if (this.isUnavailable) {
+                return "This dataset is not yet editable.";
+            }
+            return "Edit Attributes";
+        },
+        icon() {
+            const stateIcon = CONTENTSTATE[this.state] && CONTENTSTATE[this.state].icon;
+            if (stateIcon) {
+                return `fa fa-${stateIcon}`;
+            }
+            return null;
+        },
+        isUnavailable() {
+            return this.item.purged || ["upload", "new"].includes(this.state);
+        },
+    },
+    methods: {
+        onDragStart(evt) {
+            evt.dataTransfer.dropEffect = "move";
+            evt.dataTransfer.effectAllowed = "move";
+            evt.dataTransfer.setData("text", JSON.stringify([this.item]));
+        },
+        onDisplay() {
+            const id = this.item.id;
```

Comment:
So within collections this does not seem to be the HDA id, but that is what the endpoint expects. So if you click on the eye icon you either get a different dataset displayed, a permission denied error or a invalid id error. Interestingly, the dataset actions info button works.

Reply:
Yep, so you could fix it with
```suggestion
            const id = this.item.object?.id || this.item.id;
```

But I think this is an anti-pattern that deviates from the database and API model. It would be better if you determine first what the item is you're dealing with, and then you dispatch to a component that only displays HDAs or only displays collection elements.

Reply:
Thanks a lot for catching this. I fixed this by parsing the `history item` and the `collect element object` as prop. This should resolve this anti-pattern if I understand correctly.